c2rust-analyze/src/main.rs

```diff
@@ -274,10 +276,15 @@ fn is_build_script(at_args: &[String]) -> anyhow::Result<bool> {
 
 /// Run as a `rustc` wrapper (a la `$RUSTC_WRAPPER`/[`RUSTC_WRAPPER_VAR`]).
 fn rustc_wrapper() -> anyhow::Result<()> {
-    let mut at_args = env::args().skip(1).collect::<Vec<_>>();
+    let no_cargo = env::var_os(NO_CARGO_VAR).is_some();
+    let mut at_args = if no_cargo {
+        env::args().collect::<Vec<_>>()
+    } else {
+        env::args().skip(1).collect::<Vec<_>>()
```

Comment:
Could you add a sanity check that the first argument is indeed `cargo`? Is it supposed to be the path to the binary?

Reply:
The first arg will be the current executable, `c2rust-analyze`.